src/battle/battle_controller_player.c

```diff
@@ -1006,3 +1008,696 @@ void BattleControllerPlayer_UpdateFieldCondition(BattleSystem *bsys, BATTLECONTE
         ctx->command = CONTROLLER_COMMAND_10;
     }
 }
+
+typedef enum UpdateMonConditionState {
+    UMC_STATE_INGRAIN,
+    UMC_STATE_AQUA_RING,
+    UMC_STATE_ABILITY,
+    UMC_STATE_HELD_ITEM,
+    UMC_STATE_LEFTOVERS_RECOVERY,
+    UMC_STATE_LEECH_SEED,
+    UMC_STATE_POISON,
+    UMC_STATE_BAD_POISON,
+    UMC_STATE_BURN,
+    UMC_STATE_NIGHTMARE,
+    UMC_STATE_CURSE,
+    UMC_STATE_BINDING,
+    UMC_STATE_BAD_DREAMS,
+    UMC_STATE_UPROAR,
+    UMC_STATE_RAMPAGE,
+    UMC_STATE_DISABLE,
+    UMC_STATE_ENCORE,
+    UMC_STATE_LOCK_ON,
+    UMC_STATE_CHARGE,
+    UMC_STATE_TAUNT,
+    UMC_STATE_MAGNET_RISE,
+    UMC_STATE_HEALBLOCK,
+    UMC_STATE_EMBARGO,
+    UMC_STATE_YAWN,
+    UMC_STATE_HELD_ITEM_STATUS,
+    UMC_STATE_HELD_ITEM_DAMAGE,
+    UMC_STATE_END
+} UpdateMonConditionState;
+
+//static
+void BattleControllerPlayer_UpdateMonCondition(BattleSystem *bsys, BATTLECONTEXT *ctx) {
+    int i;
+    u8 flag = 0;
+    int maxBattlers;
+    int battlerId;
+    
+    maxBattlers = BattleSystem_GetMaxBattlers(bsys);
+    
+    if (ov12_0224DC74(ctx, ctx->command, ctx->command, 1) == TRUE) {
```

Comment:
does this work if you add all the following ones to a single if statement with ||?

Reply:
probably but it'll looks worse so Im not gonna change it

Reply:
entirely fair